softwerkskammer/test/mailsender/mailsenderService_test.js

```diff
@@ -327,6 +331,208 @@ describe('MailsenderService', () => {
 
   });
 
-});
+  describe('sending to contact persons of a group', () => {
+
+    function getGroups(groupIdToLookUp, fakeImplementation) {
+      return sinon.stub(groupsService, 'getGroups')
+        .withArgs([groupIdToLookUp], sinon.match.any)
+        .callsFake(fakeImplementation);
+    }
+
+    function thereIsAGroup(group) {
+      return getGroups(group.id, function (passedGroupName, callback) {
+        callback(null, [group]);
+      });
+    }
+
+    function thereIsNoGroup() {
+      return sinon.stub(groupsService, 'getGroups')
+        .callsFake(function (passedGroupId, callback) {
+          callback(null, []);
+        });
+    }
+
+    function getGroupFails(groupId) {
+      return getGroups(groupId, function (passedGroupNames, callback) {
+        callback(new Error('getGroups failed'));
+      });
+    }
+
+    function getGroupOrganizers(groupId, fakeImplementation){
+      return sinon.stub(groupsAndMembersService, 'getOrganizersOfGroup')
+        .withArgs(groupId, sinon.match.any)
+        .callsFake(fakeImplementation);
+    }
+
+    function loadingGroupOrganizersFailsFor(groupId) {
+      return getGroupOrganizers(groupId, (passedGroupName, callback) => {
+        callback(new Error('no soup for you'));
+      });
+    }
+
+    function memberWithMailAdress(email) {
+      return new Member({
+        email: email
+      });
+    }
+
+    function anyMemberWithEmail() {
+      return memberWithMailAdress('any@example.org');
+    }
+
+    function groupIsOrganizedBy(groupName, organizers) {
+      getGroupOrganizers(groupName, (passedGroupId, callback) => {
+        callback(null, organizers);
+      });
+    }
+
+    function provideValidOrganizersForGroup(groupName) {
+      groupIsOrganizedBy(groupName, [anyMemberWithEmail()]);
+    }
+
+    const groupId = 'any-group-id';
+
+    describe('when contact organizers is disabled for group', () => {
```

Comment:
Bei nur einem testcase im describe bevorzuge ich kein beforeEach, sondern direkt im "it"